packages/ui/src/hooks/use-list-translation.ts

```diff
@@ -0,0 +1,78 @@
+import { condString, conditionalArray } from '@silverhand/essentials';
+import { useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+
+/**
+ * Returns whether the given character is a CJK character.
+ *
+ * @see https://stackoverflow.com/questions/43418812
+ */
+const isCjk = (char?: string) =>
+  Boolean(
+    char?.[0] && /[\u3040-\u30FF\u3400-\u4DBF\u4E00-\u9FFF\uF900-\uFAFF\uFF66-\uFF9F]/.test(char[0])
+  );
+
+/**
+ * Returns a function that translates a list of strings into a human-readable list. If the list
+ * is empty, `undefined` is returned.
+ *
+ * For non-CJK languages, the list is translated with the Oxford comma. For CJK languages, the
+ * list is translated with the AP style.
+ *
+ * CAUTION: This function may not be suitable for translating lists of non-English strings if the
+ * target language does not have the same rules for list translation as English.
+ *
+ * @example
+ * ```ts
+ * const translateList = useListTranslation();
+ *
+ * // en
+ * translateList([]); // undefined
+ * translateList(['a']); // 'a'
+ * translateList(['a', 'b']); // 'a and b'
+ * translateList(['a', 'b', 'c']); // 'a, b, or c'
+ * translateList(['a', 'b', 'c'], 'and'); // 'a, b, and c'
+ *
+ * // zh
+ * translateList(['a', 'b']); // 'a 或 b'
+ * translateList(['苹果', '橘子', '香蕉']); // '苹果、橘子或香蕉'
+ * translateList(['苹果', '橘子', 'banana']); // '苹果、橘子或 banana'
+ * ```
+ */
+const useListTranslation = () => {
```

Comment:
Is there any limitations to the default i18n [list formatting](https://www.i18next.com/translation-function/formatting#list)?

Reply:
cock, i tried but didn't notice this at all